tests/managers_regress/tests.py

```diff
@@ -156,6 +156,18 @@ def test_field_can_be_called_exact(self):
         relation = related.test_fk.create()
         self.assertEqual(related.test_fk.get(), relation)
 
+    @isolate_apps('managers_regress')
+    def test_generated_manager_does_not_override(self):
+        # Make sure a generated manager does not override an attribute
+        # defined explicitly on the class
+        msg = (
+            "Class Bogus has an attribute 'objects' which would be "
+            "overridden. Check definitions."
+        )
+        with self.assertRaisesMessage(AttributeError, msg):
+            class Bogus(models.Model):
+                objects = "not a manager"
+
```

Comment:
Hey @shaib. 

From our original discussion, 🌔s-ago :), I wondered if we weren't going to check the manager type in this `objects` case... 🤔 

As you have it, if I make the bad assignment on a parent class, it doesn't raise: 

```
        msg = (
            "Class Bogus has an attribute 'objects' which would be "
            "overridden. Check definitions."
        )
        with self.assertRaisesMessage(AttributeError, msg):
            class BogusBase(models.Model):
                objects = "not a manager"

                class Meta:
                    abstract = True

            class Bogus(BogusBase):
                ...
```

Maybe that's fine. Can I just ask your thought? 
Thanks. 😉

Reply:
Hmmmm.
On one hand -- that does not look like desired behavior to me.
On the other -- that is quite orthogonal to what the patch is trying to do.
On the third -- this does go directly to the point of the ticket that the patch is a suggested solution for.

I think the patch is an improvement to Django, beyond the ticket; maybe we should just change its commit-message from `Fixed #32640` to `Refs #32640`?